src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -357,8 +357,8 @@ RNN::RNN(const std::shared_ptr<ov::Node>& op, const dnnl::engine& eng, WeightsSh
         initSequence();
     }
 
-    inDataTypes.reserve(getOriginalInputsNumber());
-    outDataTypes.reserve(getOriginalOutputsNumber());
+    inDataTypes.resize(getOriginalInputsNumber());
+    outDataTypes.resize(getOriginalOutputsNumber());
```

Comment:
why do we need change to resize here?

Reply:
reserve does not change the length of the vector.
And since we are setting the values of the inDataType and outDataType using index (not push_back) we have to resize, not reserve.